### DISCUSSION TITLE
How to show, hide field in "uncontrolled" form mode

### DISCUSSION THREAD

**Discussion by xauuu** [mantinedev/mantine#7060] [Q&A]
Hello Mantine Community!

I want my form to show or hide a certain field when the input changes in an uncontrolled form.
```
{form.getValues().authentication === 'Bearer' && (
  <Textarea label='Token' key={form.key('token')} {...form.getInputProps('token')} />
)}
```
It works in controller mode, but uncontrolled doesn't 

Thanks!

**Comment by rtivital** [comment]
You can subscribe to values changes with form.watch – https://mantine.dev/form/values/#formwatch

**Reply by xauuu** [reply]
and store the value in useState right?

**Reply by xauuu** [reply]
And I have another question
![image](https://github.com/user-attachments/assets/a8304109-f2d3-4cd8-b98c-f26fb1bccb3c)
![image](https://github.com/user-attachments/assets/0829bd16-8de3-48ab-a49b-e66b9475537a)
I use Select and add 1 option using react-query's mutate api, after success I use `queryClient.invalidateQueries` to get back the `data` of Select, and use `form.setFieldValue` the newly added option, it is like the image above, not updating the label of Select, but in the `data` it has updated and selected.
It is in uncontrolled mode, and in controlled mode it is works normally

**Reply by rtivital** [reply]
Make sure that you've set `key={form.key('field-name')}` like it is shown in the documentation

**Reply by xauuu** [reply]
When I successfully create a new one, and refetch the list, the label is updated, but when I select one then update it and refetch it, it doesn't update the label
And I have set key for select
![image](https://github.com/user-attachments/assets/f222d074-ae69-446d-8e29-fd57b0f3385f)
```
  const handleSavePushConfig = (variables: PushConfigSet) => {
    upsertPushConfigSet.mutateAsync(
      {
        variables
      },
      {
        onSuccess: async ({ message, data }) => {
          await queryClient.invalidateQueries({ queryKey: [DATABASE_KEY.PUSH_CONFIG_SET.LIST] })
          showCustomNotification({ message: message, type: 'success' })
          form.setFieldValue('push_config_set_id', data)
          handlerOpenPush.close()
        },
        onError: (error) => {
          showCustomNotification({ message: error.message, type: 'error' })
        }
      }
    )
  }
```

**Comment by hitbear518** [comment]
Will there be a watch api like react-hook-form to subscribe values when using uncontrolled mode in the future?

**Reply by rtivital** [reply]
What do you mean? form.watch already provides this feature

**Reply by hitbear518** [reply]
form.watch has a callback, I'm using a useState with form.watch to get the latest value of fields.

The api I'm referring to is like `const someFiledVlaue = watch('someField');` So that I can render that value within React Component

**Reply by rtivital** [reply]
If you need this feature, use controlled mode and `form.getValues()`, there won't be any difference in performance

**Reply by hitbear518** [reply]
got it, controlled mode and form.values will do it. 

only big forms may have difference on perf
